core/src/consensus/light_client/verification.rs

```diff
@@ -14,73 +14,142 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use ccrypto::blake256;
+use super::{CoreHeader, LightHeader, Seal};
+use ckey::verify_schnorr;
 use ctypes::CompactValidatorSet;
 use merkle_trie::proof::{verify, CryptoProof, CryptoProofUnit};
+use primitives::{Bytes, H256};
 
-use super::LightHeader;
-use super::Seal;
+pub fn verify_signature(block_hash: H256, vset: &CompactValidatorSet, seal: &Seal) -> bool {
+    let n = vset.0.len();
```

Comment:
Please do not use 0 and 1.

Reply:
Have you considered this?
https://doc.rust-lang.org/rust-by-example/generics/new_types.html

Reply:
Let's use the 0 only for the new type.

Reply:
Here, that's the case.

Reply:
I'll fix up other tuples then.

Reply:
Well there are some, but I have no clue how to fix them. Any suggestion?

Reply:
I applied destructuring assignment for most case, but I'll just left very short (e.g closure) part.

Reply:
Let's destructure the tuple.
https://github.com/CodeChain-io/foundry/pull/172#discussion_r379322726